spikesorters/kilosort2/kilosort2_new.py

```diff
@@ -0,0 +1,169 @@
+from pathlib import Path
+import os
+import sys
+import numpy as np
+from typing import Union
+
+import spikeextractors as se
+from ..basesorter import BaseSorter
+from ..sorter_tools import _call_command_split
+
+
+def check_if_installed(kilosort2_path: Union[str, None]):
+    if kilosort2_path is None:
+        return False
+    assert isinstance(kilosort2_path, str)
+
+    if kilosort2_path.startswith('"'):
+        kilosort2_path = kilosort2_path[1:-1]
+    kilosort2_path = str(Path(kilosort2_path).absolute())
+
+    if (Path(kilosort2_path) / 'master_kilosort.m').is_file():
+        return True
+    else:
+        return False
+
+
+class Kilosort2SorterNew(BaseSorter):
+    """
+    """
+
+    sorter_name: str = 'kilosort2'
+    kilosort2_path: Union[str, None] = os.getenv('KILOSORT2_PATH', None)
+
+    _default_params = {
+        'detect_threshold': 5,
+        'electrode_dimensions': None,
+        'car': True,
+        'minFR': 0.1,
+    }
+
+    installation_mesg = """\nTo use Kilosort run:\n
+        >>> git clone https://github.com/cortex-lab/KiloSort
+    and provide the installation path by setting the KILOSORT2_PATH
+    environment variables or using Kilosort2SorterNew.set_kilosort2_path().\n\n
+
+    More information on KiloSort2 at:
+        https://github.com/MouseLand/Kilosort2
+    """
+
+    def __init__(self, **kargs):
+        BaseSorter.__init__(self, **kargs)
+
+    @staticmethod
+    def set_kilosort2_path(kilosort2_path: str):
+        Kilosort2SorterNew.kilosort2_path = kilosort2_path
+
+    def set_params(self, **params):
+        BaseSorter.set_params(self, **params)
+
+    def _setup_recording(self, recording, output_folder):
+
+        source_dir = Path(Path(__file__).parent)
+
+        p = self.params
+
+        if not check_if_installed(Kilosort2SorterNew.kilosort2_path):
+            raise Exception(Kilosort2SorterNew.installation_mesg)
+        assert isinstance(Kilosort2SorterNew.kilosort2_path, str)
+
+        # prepare electrode positions
+        electrode_dimensions = p['electrode_dimensions']
+        if electrode_dimensions is None:
+            electrode_dimensions = [0, 1]
+        if 'group' in recording.get_channel_property_names():
+            groups = [recording.get_channel_property(
```

Comment:
you can also use:

`recording.get_channel_groups()`

Reply:
ok